src/common/types/bit.cpp

```diff
@@ -146,6 +147,44 @@ string Bit::ToBit(string_t str) {
 	return output_str.GetString();
 }
 
+void Bit::BlobToBit(string_t blob, string_t &output_str) {
+	auto data = const_data_ptr_cast(blob.GetData());
+	auto output = output_str.GetDataWriteable();
+	idx_t size = blob.GetSize();
+
+	*output = 0; // No padding
+	for (idx_t idx = 0; idx < size; ++idx) {
+		output[1 + idx] = data[idx];
+	}
+}
+
+string Bit::BlobToBit(string_t blob) {
+	auto buffer = make_unsafe_uniq_array<char>(blob.GetSize() + 1);
+	string_t output_str(buffer.get(), blob.GetSize() + 1);
+	Bit::BlobToBit(blob, output_str);
+	return output_str.GetString();
+}
+
+void Bit::BitToBlob(string_t bit, string_t &output_blob) {
+	D_ASSERT(bit.GetSize() == output_blob.GetSize() + 1);
+
+	auto data = const_data_ptr_cast(bit.GetData());
+	auto output = output_blob.GetDataWriteable();
+	idx_t size = output_blob.GetSize();
+
+	output[0] = data[1] & ((1 << (8 - data[0])) - 1);
```

Comment:
Could you move this into a helper function?
Something like `data_t Bit::GetFirstByte(const_data_ptr_t bitstring_data);`

It took me a little bit to realize what was going on here, so it might be better to have some comments around it regardless.
And it's also used in multiple places, which is also a reason to move into a function

Reply:
Makes sense. Moved it into a function.